### PR TITLE
Make the map type inaccessible to users

### PR DESCRIPTION
**Pull request by dominiklohmann**
We want to remove the map type, but doing that is a larger operation. To get the user-facing breaking change out earlier, i.e., in time for 3.0, we're just making it inaccessible first and are then removing it later on.

Additionally, this PR removes the only place in VAST that ever used the map type for table slices written to disk: The `vast.metrics` event, whose metadata was never queryable.

Now, behold the beauty of actually queryable metrics from the self-sink:

```
$ vast -qq export json '#type == "vast.metrics.passive-store.init.runtime"
  | select ts,value
  | summarize min(value), max(value) by ts resolution 10s'
{"ts": "2023-02-28T17:21:50.000000", "min(value)": 0.218875, "max(value)": 107.280125}
{"ts": "2023-02-28T17:20:00.000000", "min(value)": 0.549292, "max(value)": 0.549292}
```

To enable the self-sink for testing, set the following:

```yaml
# <prefix>/etc/vast/vast.yaml
vast:
  enable-metrics: true
  metrics:
    self-sink:
      enable: true
```

```[tasklist]
### Tasks
- [x] Make the map type inaccessible
- [x] Avoid using the map type for metrics
- [x] Update the VAST v3.0 blog post draft
- [x] ~Update the documentation~ @mavam will do that in tenzir/vast#2967 for me
- [x] Add changelog entry
```

### CODE REVIEW

**Comment by mavam** [comment]
Loving the "self sink" example!

File: web/docs/setup/monitor.md [hunk 1]
```diff
@@ -71,14 +71,50 @@ vast:
...
+      enable: false
 ```
 
-Both sinks write metrics as NDJSON and inlines the `metadata` key-value pairs
-into the top-level object. VAST buffers metrics for these sinks in order to
-batch I/O operations. To enable real-time metrics reporting, set the options
-`vast.metrics.file-sink.real-time` or `vast.metrics.uds-sink.real-time`
+Both the file and UDS sinks write metrics as NDJSON and inlines the `metadata`
+key-value pairs into the top-level object. VAST buffers metrics for these sinks
+in order to batch I/O operations. To enable real-time metrics reporting, set the
```
Comment (mavam) [inline web/docs/setup/monitor.md]:
```suggestion
to batch I/O operations. To enable real-time metrics reporting, set the
```

File: web/docs/setup/monitor.md [hunk 2]
```diff
@@ -71,14 +71,50 @@ vast:
...
+    - <metadata...>
+```
+
+Here's an example that queries for the start up latency of VAST's stores,
+grouping into 10 second buckets and looking at the minimum and the maximum
+latency, respectively, for all buckets.
+
+```bash
+vast export json '#type == "vast.metrics.passive-store.init.runtime"
+  | select ts,value
```
Comment (mavam) [inline web/docs/setup/monitor.md:107]:
```suggestion
  | select ts, value
```

File: web/docs/setup/monitor.md [hunk 3]
```diff
@@ -71,14 +71,50 @@ vast:
...
+  record:
+    - ts: timestamp
+    - version: string
+    - actor: string
+    - key: string
+    - value: string
+    - <metadata...>
+```
+
+Here's an example that queries for the start up latency of VAST's stores,
```
Comment (mavam) [inline web/docs/setup/monitor.md:101]:
```suggestion
Here's an example that shows the start up latency of VAST's stores,
```

File: web/docs/setup/monitor.md [hunk 4]
```diff
@@ -71,14 +71,50 @@ vast:
...
+grouping into 10 second buckets and looking at the minimum and the maximum
```
Comment (mavam) [inline web/docs/setup/monitor.md:102]:
```suggestion
grouped into 10 second buckets and looking at the minimum and the maximum
```

File: libvast/src/system/accountant.cpp [hunk 5]
```diff
@@ -71,7 +71,7 @@ struct accountant_state_impl {
   std::unordered_map<caf::actor_id, std::string> actor_map;
 
   /// Stores the builder instance.
```
Comment (mavam) [inline libvast/src/system/accountant.cpp:73]:
```suggestion
  /// Stores the builder instance per type name.
```

File: web/blog/vast-v3.0/index.md [hunk 6]
```diff
@@ -5,13 +5,12 @@ description: The VAST Language Evolution — Dataflow Pipelines
 authors: [dominiklohmann, dakostu]
 date: 2023-02-15
 image: /path/to/image.png
-tags: [release, vastql, cef, performance, introspection, regex]
+tags: [release, vast-language, cef, performance, introspection, regex]
```
Comment (mavam) [inline web/blog/vast-v3.0/index.md:8]:
```suggestion
tags: [release, language, cef, performance, introspection, regex]
```

File: web/docs/setup/monitor.md [hunk 7]
```diff
@@ -71,14 +71,50 @@ vast:
...
-Both sinks write metrics as NDJSON and inlines the `metadata` key-value pairs
-into the top-level object. VAST buffers metrics for these sinks in order to
-batch I/O operations. To enable real-time metrics reporting, set the options
+Both the file and UDS sinks write metrics as NDJSON and inline the `metadata`
+key-value pairs into the top-level object. VAST buffers metrics for these sinks
+to batch I/O operations. To enable real-time metrics reporting, set the options
 `vast.metrics.file-sink.real-time` or `vast.metrics.uds-sink.real-time`
 respectively in your configuration file.
 
+:::tip Self Sink ❤️ Pipelines
```
Comment (mavam) [inline web/docs/setup/monitor.md:85]: The spacing around ❤️ is different here. </bikeshed off>

File: web/docs/setup/monitor.md [hunk 8]
```diff
@@ -71,14 +71,50 @@ vast:
...
+The self-sink routes metrics as events into VAST's internal storage engine,
```
Comment (mavam) [inline web/docs/setup/monitor.md:86]: NB: "self sink" is a terrible name, but I don't have a better one at hand.

File: web/docs/setup/monitor.md (hunk 7 above)
Comment (dominiklohmann) [reply web/docs/setup/monitor.md:85]: Fixed this in https://github.com/tenzir/vast/pull/2976/commits/f594483ae0a48c67bb844cee77dc7294b781aaf6 already.

File: web/docs/setup/monitor.md (hunk 1 above)
Comment (dominiklohmann) [reply web/docs/setup/monitor.md]: Fixed this in https://github.com/tenzir/vast/pull/2976/commits/f594483ae0a48c67bb844cee77dc7294b781aaf6 already.

**Pull request merged**